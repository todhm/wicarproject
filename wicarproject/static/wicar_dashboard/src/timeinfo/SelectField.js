import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import {connect } from 'react-redux'
import {changeAvailabilityTime} from './action'



const SelectField=(props)=>{

        const{optionList,defaultLabel,className,name,index,changeAvailability,value,label,onChange} = props
        return(
                <span className={className}>
                    <select name={name} className="required" value = {value} onChange={onChange} >
                        {optionList.map((option,index)=>(
                             <option value={option.value} key={option.value}>{option.label}</option>
                        ))}
                    </select>
                    <span className="text">{label}</span>
                </span>
        )
    }




export default SelectField;
