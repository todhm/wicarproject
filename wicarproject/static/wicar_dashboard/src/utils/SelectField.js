import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import {TextField,MenuItem} from '@material-ui/core';
import Select from '@material-ui/core/Select';


const  SelectField=(props)=>{
    const {name,label,value,onChange,error,defaultLabel,optionList,labelEqual,noDefault} = props
    return(
        <TextField
            select
            fullWidth={true}
            name={name}
            label={label}
            value={value}
            onChange={onChange}
            helperText={error}
            error={error?true:false}
            margin="normal"
            >
        {noDefault?
            null:
            <option value="">{defaultLabel}</option>
        }
          {optionList.map((option)=>(
                  labelEqual?
                  <option value={option.codeName} key={option.codeName}>{option.codeName}</option>
                  :<option value={option.value} key={option.value}>{option.label}</option>
          ))}
      </TextField>
    )
}


export default SelectField;
