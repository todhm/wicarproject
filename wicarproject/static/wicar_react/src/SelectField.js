import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'


class SelectField extends Component{

    render(){
        return(
            <div className="grid-item grid-item--3">
                <label htmlFor="js-makesInput">{this.props.label}</label>
                <span className="styled-select-container styled-select-container--fluid">
                    <select id="js-makesInput" name={this.props.name} className="required" value = {this.props.value}  onChange={this.props.onChange} >
                        <option value="">{this.props.defaultLabel}</option>
                        {this.props.optionList.map((option)=>(
                             <option value={option.codeName} key={option.codeName}>{option.codeName}</option>
                        ))}
                    </select>
                    {this.props.value===""?
                        <span className="text brand-select">{this.props.defaultLabel}</span>:
                        <span className="text">{this.props.value}</span>
                        }
                        <span className="errorMessage-text js-locationErrorText">{this.props.error}</span>

                </span>
            </div>
        )
    }
}


export default SelectField;
