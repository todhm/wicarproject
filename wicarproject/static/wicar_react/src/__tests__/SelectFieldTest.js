import React from 'react';
import {shallow,mount} from 'enzyme';
import SelectField from '../SelectField.js'
import	renderer	from	'react-test-renderer';

const optionList = [{codeName:"Jordan"},{codeName:"Jordan2"}]
const onChange = jest.fn()

test('Test SelectField Render Correctly', ()	=>	{

    const   wrapper	=	shallow(
            <SelectField
                optionList={optionList}
                onChange={onChange}

                />
    );
    const label = wrapper.find('label');
    const renderedOptions = wrapper.find('option');
    expect(label.length).toBe(1)
    expect(renderedOptions.length).toBe(3)
    optionList.forEach((option,index)=>{
        expect(renderedOptions.get(index + 1).props.children).toBe(option.codeName)
    })

});


describe('When not authenticated', () => {
    const   component =<SelectField
                optionList={optionList}
                onChange={onChange}
                />
    it(' Select renders properly', () => {
        const wrapper = shallow(component);
        const selectField = wrapper.find('select');
        expect(selectField.length).toBe(1);
        const label = wrapper.find('label');
        const renderedOptions = wrapper.find('option');
        expect(label.length).toBe(1)
        expect(renderedOptions.length).toBe(3)
        optionList.forEach((option,index)=>{
            expect(renderedOptions.get(index + 1).props.children).toBe(option.codeName)
        })
    });
    it('SelectField renders a snapshot properly', () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
        });

});
