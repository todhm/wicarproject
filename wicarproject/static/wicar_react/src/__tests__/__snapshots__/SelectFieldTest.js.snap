// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When not authenticated SelectField renders a snapshot properly 1`] = `
<div
  className="grid-item grid-item--3"
>
  <label
    htmlFor="js-makesInput"
  />
  <span
    className="styled-select-container styled-select-container--fluid"
  >
    <select
      className="required"
      id="js-makesInput"
      name={undefined}
      onChange={[Function]}
      value={undefined}
    >
      <option
        value=""
      />
      <option
        value="Jordan"
      >
        Jordan
      </option>
      <option
        value="Jordan2"
      >
        Jordan2
      </option>
    </select>
    <span
      className="text"
    />
    <span
      className="errorMessage-text js-locationErrorText"
    />
  </span>
</div>
`;
